pkg/operator/controller/controller.go

```diff
@@ -41,157 +30,3 @@ const (
 type Controller interface {
 	Register(context.Context, manager.Manager) error
 }
-
-type SingletonBuilder struct {
-	mgr  manager.Manager
-	name string
-}
-
-func NewSingletonManagedBy(m manager.Manager) SingletonBuilder {
-	return SingletonBuilder{
-		mgr: m,
-	}
-}
-
-func (b SingletonBuilder) Named(name string) SingletonBuilder {
-	b.name = name
-	return b
-}
-
-func (b SingletonBuilder) Complete(r reconcile.Reconciler) error {
-	return b.mgr.Add(newSingleton(r, b.name))
-}
-
-type Singleton struct {
-	reconcile.Reconciler
-	name        string
-	rateLimiter ratelimiter.RateLimiter
-}
-
-func newSingleton(r reconcile.Reconciler, name string) *Singleton {
-	s := &Singleton{
-		Reconciler:  r,
-		name:        name,
-		rateLimiter: workqueue.DefaultItemBasedRateLimiter(),
-	}
-	s.initMetrics()
-	return s
-}
-
-// initMetrics is effectively the same metrics initialization function used by controller-runtime
-// https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/internal/controller/controller.go
-func (s *Singleton) initMetrics() {
```

Comment:
I assume all of these metrics are injected since we're now using the controller-runtime controller directly?

Reply:
Yep, these are added for free by controller-runtime. I validated this through checking prometheus after the change,